File: backend/app/schemas/audit.py
from typing import Any, Dict, Optional
from pydantic import BaseModel, Field
from backend.app.schemas.common import AuditActorType


class AuditLogCreate(BaseModel):
    action: str
    actor_type: AuditActorType
    actor_name: str
    entity_type: str
    entity_id: str
    entity_name: str
    status: str = "SUCCESS"
    details: Dict[str, Any] = Field(default_factory=dict)
    ip_address: Optional[str] = "127.0.0.1"


class AuditLogResponse(BaseModel):
    id: str
    timestamp: str
    action: str
    actor_type: AuditActorType
    actor_name: str
    entity_type: str
    entity_id: str
    entity_name: str
    status: str
    details: Dict[str, Any] = Field(default_factory=dict)
    ip_address: Optional[str] = None

File: frontend/src/types/audit.ts
import { UUID } from './common'

export type AuditAction =
  | 'SIMULATION_STARTED'
  | 'SIMULATION_COMPLETED'
  | 'ATTACK_PLAN_GENERATED'
  | 'ATTACK_PLAN_VALIDATED'
  | 'BYPASS_DETECTED'
  | 'VULNERABILITY_DISCOVERED'
  | 'EXPLANATION_GENERATED'
  | 'PATCH_GENERATED'
  | 'PATCH_VALIDATED'
  | 'PATCH_SIMULATED'
  | 'PATCH_APPROVED'
  | 'PATCH_REJECTED'
  | 'POLICY_CREATED'
  | 'POLICY_VERSION_DEPLOYED'
  | 'BENCHMARK_EXECUTED'
  | 'USER_LOGIN'

export interface AuditLogEntry {
  id: UUID
  timestamp: string
  action: AuditAction
  actorType: 'USER' | 'SYSTEM' | 'AI_AGENT'
  actorName: string
  entityType: 'POLICY' | 'SIMULATION' | 'VULNERABILITY' | 'PATCH' | 'BENCHMARK' | 'DATASET'
  entityId: UUID
  entityName: string
  status: 'SUCCESS' | 'WARNING' | 'FAILED' | 'REJECTED'
  details: Record<string, any>
  ipAddress?: string
}
